### PR TITLE
[helper-plugin]: migrate Form to TypeScript

### PR DESCRIPTION
**Pull request by Tegacreatives**
### What does it do?

Migrate Form.jsx helper-plugin component to Typescript

### Related issue(s)/PR(s)

[#17690](https://github.com/strapi/strapi/issues/17690)

### CODE REVIEW

File: packages/core/helper-plugin/src/components/Form.tsx [hunk 1]
```diff
@@ -2,23 +2,23 @@ import React, { useEffect, useRef } from 'react';
 
 import { Form as FormikForm, useFormikContext } from 'formik';
 
-const Form = (props) => {
-  const formRef = useRef(null);
+const Form = ({ ...props }) => {
+  const formRef = useRef<HTMLFormElement>(null);
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/Form.tsx:6]: we know `formRef` will be defined so we can do:

```suggestion
  const formRef = useRef<HTMLFormElement>(null!);
```

Which will remove your unnecessary optional chaining

File: packages/core/helper-plugin/src/components/Form.tsx [hunk 2]
```diff
@@ -2,23 +2,23 @@ import React, { useEffect, useRef } from 'react';
 
 import { Form as FormikForm, useFormikContext } from 'formik';
 
-const Form = (props) => {
-  const formRef = useRef(null);
+const Form = ({ ...props }) => {
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/Form.tsx:5]: @markkaylor do we use this anywhere? It's really bad code imo, i'm wondering if it's best we deprecate this and remove it's usage – what do you think?

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/helper-plugin/src/components/Form.tsx (hunk 2 above)
Comment (joshuaellis) [reply packages/core/helper-plugin/src/components/Form.tsx:5]: i think deprecate first cause then everyone will see it and can action it as they go 🤓

Comment (markkaylor) [reply packages/core/helper-plugin/src/components/Form.tsx:5]: Sounds good to me. @Tegacreatives could you add an `@deprecated` annotation with a message to use Formik Form directly instead.

File: packages/core/helper-plugin/src/components/Form.tsx (hunk 1 above)
Comment (Tegacreatives) [reply packages/core/helper-plugin/src/components/Form.tsx:6]: Thanks a lot @joshuaellis

File: packages/core/helper-plugin/src/components/Form.tsx (hunk 2 above)
Comment (Tegacreatives) [reply packages/core/helper-plugin/src/components/Form.tsx:5]: Sure @markkaylor

**Comment by markkaylor** [comment]
@Tegacreatives Your pr is targeting the wrong branch, could you please reopen based off of [chore/ts-migrate-helper-plugin](https://github.com/strapi/strapi/tree/chore/ts-migrate-helper-plugin?rgh-link-date=2023-08-17T14%3A04%3A55Z) as stated in [the issue description](https://github.com/strapi/strapi/issues/17690)

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**